ingredient/ingredient-app-insights/src/functions.ts

```diff
@@ -1,12 +1,25 @@
-import {BaseUtility, IngredientManager} from '@azbake/core'
+import { BaseUtility, IngredientManager } from '@azbake/core'
+import { ApplicationInsightsManagementClient } from '@azure/arm-appinsights'
 
 export class AppInsightsUtils extends BaseUtility {
-    
+
     public create_resource_name(): string {
         let util = IngredientManager.getIngredientFunction("coreutils", this.context)
 
         const name = util.create_resource_name("ai", null, true);
         return name;
     }
-}
 
+    public async get_instrumentation_key(appInsightsName: string, rg: string): Promise<string> {
```

Comment:
RG should be optional, and if not set we should generate the RG name for the shared app insight domain RG.

Reply:
I was planning to use the get_resource_group(packageName) function in conjunction with get_instrumentation_key.  Handling it in the recipe rather than passing it as a param to get_instrumentation_key.  I may be thinking about it wrong though.  Thoughts?

Reply:
couple of things:
1) Remember that other recipes are going to be using this. So you can't create the RG based on that package name (it won't match the shared AI resource group for all AI domain instances). Your util function needs to hard code the RG name generation logic (using current region code, current env code, etc, and probably a hard coded name) so that it can resolve no matter what package it runs from.
2) I would not do that external to this util method and pass it in. That's adding an extra required step to developers who want to fetch AI keys, and if they don't do it.. their package is broken.

Reply:
I believe I would need to add a param for the name of the package that created the shared app insights domain RG in order to make RG an optional param.  

I was planning to use the get_resource_group(package name) function within the recipe to look up the resource group and pass it to get_instrumentation key.  Thoughts?

Reply:
Sorry for the delay.  

I updated the ingredient to now accept an optional resource group.  If not supplied, it generates the resource group name.  The AzurePlatform repo would generate the name of the App Insights resource like this -
appInsightsName: "[appinsights.create_resource_name('Wounds')]"

The developer would then get the iKey in their application's recipe like this.
iKey: "[appinsights.get_instrumentation_key('Wounds')]"

Reply:
Hey Bill, wondering if I need to rethink the approach for resource groups.  It looks like the region code is appended when util.resource_group() is called but I'm planning to only have App Insights in the US East region for centralization.  Thoughts?

Reply:
Apologies... I realized that I didn't push my commit to remote.  It's there now.

Reply:
You need to make sure that the RG name creation for the AI RG is stable for any recipe that uses it. How you have it setup right now each recipe will generate a unique RG name since create_resource_name embeds the recipe short name into it. If you are hard coding EUS, just need to code that into a proper RG creation util for the AI RG.

Reply:
I'd like to get your thoughts on the current set of changes.  I'm still requiring a resource group param is passed due to the current design of a single App Insights recipe containing multiple App Insights ingredients (per domain).  In the readme, I included examples of how the resources will be created and be referenced in other recipes.  No problem on making it optional if needed but I think it may be clearer that the App Insights resource group isn't the same one as the referencing recipe if I require it.  Thoughts?